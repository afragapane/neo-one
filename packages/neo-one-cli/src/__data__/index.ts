export * from './getClients';
export * from './getContracts';
