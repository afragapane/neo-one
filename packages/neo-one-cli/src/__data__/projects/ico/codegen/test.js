/* @hash a6c11fb7302ee32fe517aae06bcedbe0 */
// tslint:disable
/* eslint-disable */
import { createWithContracts } from '@neo-one/smart-contract-test';
import * as path from 'path';

export const withContracts = createWithContracts([
  { name: 'Escrow', filePath: path.resolve(__dirname, '../contracts/Escrow.ts') },
  { name: 'Token', filePath: path.resolve(__dirname, '../contracts/Token.ts') },
  { name: 'ICO', filePath: path.resolve(__dirname, '../contracts/ICO.ts') },
]);
