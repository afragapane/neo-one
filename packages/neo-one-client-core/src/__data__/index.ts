export * from './keys';
export * from './data';
