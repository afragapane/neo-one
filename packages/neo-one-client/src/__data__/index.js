/* @flow */
export { default as keys } from './keys';
export { default as transactions } from './transactions';
