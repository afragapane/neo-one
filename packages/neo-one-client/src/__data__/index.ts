export * from './keys';
export * from './data';
export * from './transactions';
export * from './factory';
