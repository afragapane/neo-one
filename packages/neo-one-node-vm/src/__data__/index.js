/* @flow */
export { default as createBlockchain } from './createBlockchain';
export { default as keys } from './keys';
export * as transactions from './transactions';
export * as testUtils from './utils';
