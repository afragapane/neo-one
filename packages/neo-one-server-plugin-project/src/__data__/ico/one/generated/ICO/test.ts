// tslint:disable
import { setupContractTest, SetupTestResult } from '@neo-one/smart-contract-compiler';
import * as path from 'path';
import { ICOSmartContract } from './types';

export const setupICOTest = async (): Promise<SetupTestResult<ICOSmartContract>> =>
  setupContractTest<ICOSmartContract>(path.resolve(__dirname, '../../contracts/ICO.ts'), 'ICO');
