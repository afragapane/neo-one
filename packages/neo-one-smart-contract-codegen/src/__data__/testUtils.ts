import { ABI } from '@neo-one/client-common';
import { genSmartContractTypes } from '../types';

const testABI = (abi: ABI, name: string) => {
  test(name, () => {
    expect({
      inputABI: abi,
      types: genSmartContractTypes(name, abi, false),
    }).toMatchSnapshot();
  });
};

export const testUtils = {
  testABI,
};
