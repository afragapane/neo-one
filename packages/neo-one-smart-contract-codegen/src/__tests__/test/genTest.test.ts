import { contractsPaths } from '../../__data__/contractsPaths';
import { genTest } from '../../test';

describe('genTest', () => {
  test('Token', () => {
    expect(
      genTest({
        contractsPaths,
        commonTypesPath: '/foo/bar/one/generated/types.js',
        testPath: '/foo/bar/one/generated/test.js',
      }),
    ).toMatchSnapshot();
  });
});
