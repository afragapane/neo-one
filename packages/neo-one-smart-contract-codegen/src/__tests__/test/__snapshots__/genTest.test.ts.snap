// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genTest Token 1`] = `
"
import { setupContractTest, SetupTestResult } from '@neo-one/smart-contract-test';
import * as path from 'path';
import { TokenSmartContract } from './types';

export const setupTokenTest = async (): Promise<SetupTestResult<TokenSmartContract>> =>
  setupContractTest<TokenSmartContract>(
    path.resolve(__dirname, '../../contracts/Token.ts'),
    'Token',
  );
"
`;
